libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/camera/SimpleCamera.java

```diff
@@ -59,20 +62,36 @@ public double zoom(RouteInformation routeInformation) {
     return CAMERA_ZOOM;
   }
 
+  @Override
+  public List<Point> overview(RouteInformation routeInformation) {
+    boolean invalidCoordinates = routeCoordinates == null || routeCoordinates.isEmpty();
+    if (invalidCoordinates) {
+      if (routeInformation.route() != null) {
+        setupLineStringAndBearing(routeInformation.route());
+      } else if (routeInformation.routeProgress() != null) {
+        setupLineStringAndBearing(routeInformation.routeProgress().directionsRoute());
+      }
+    }
+    return routeCoordinates;
+  }
+
   private void setupLineStringAndBearing(DirectionsRoute route) {
     if (initialRoute != null && route.equals(initialRoute)) {
       return; //no need to recalculate these values
     }
     initialRoute = route;
-
-    lineString = LineString.fromPolyline(route.geometry(), Constants.PRECISION_6);
+    routeCoordinates = generateRouteCoordinates(route);
     initialBearing = TurfMeasurement.bearing(
-      Point.fromLngLat(
-        lineString.coordinates().get(0).longitude(), lineString.coordinates().get(0).latitude()
-      ),
-      Point.fromLngLat(
-        lineString.coordinates().get(1).longitude(), lineString.coordinates().get(1).latitude()
-      )
+      Point.fromLngLat(routeCoordinates.get(0).longitude(), routeCoordinates.get(0).latitude()),
+      Point.fromLngLat(routeCoordinates.get(1).longitude(), routeCoordinates.get(1).latitude())
     );
   }
+
+  private List<Point> generateRouteCoordinates(DirectionsRoute route) {
+    if (route != null) {
+      LineString lineString = LineString.fromPolyline(route.geometry(), Constants.PRECISION_6);
+      return lineString.coordinates();
+    }
+    return null;
```

Comment:
We should consider returning a [Null Object](https://en.wikipedia.org/wiki/Null_Object_pattern#Java) instead or even better try to getting rid of it at all (maybe returning an empty list?). If not, code becomes defensive and full of `null` checks.

👀 [Null References: The Billion Dollar Mistake](https://www.infoq.com/presentations/Null-References-The-Billion-Dollar-Mistake-Tony-Hoare)

Reply:
Actually if the `route` in `setupLineStringAndBearing` is `null` L#85 will cause a NPE right?

Reply:
In any case, guard clauses normally are at the beginning of the method representing the unexpected behavior.

Reply:
Also saw the same exact code in `DynamicCameraTest.java` (`generateRouteCoordinates`). Should be abstracted and accessible to the implementation classes?